elfloader.c

```diff
@@ -91,8 +97,13 @@ int main(int argc, char **argv)
 
     entrypoint_t ep = (entrypoint_t) epaddr;
 
-    printf("Binary entrypoint is %x; invoking %p\n", ctx.ehdr.e_entry, ep);
+    printf("Binary entrypoint is %lx; invoking %p\n", ctx.ehdr.e_entry, ep);
 
     go(ep);
+
+    fclose(f);
+
+    free(buf);
```

Comment:
I'm OK with these but am generally not concerned about memory leaks in example problems (also, the program is about to exit which will free all resources anyway)

Reply:
Just a pet peeve of mine because I frequently test programs with valgrind.

Reply:
It's always a good idea to free resources, especially if you have a habit of reusing code. Not adding those lines could result in memory leaks later on, especially on embedded systems.